java/apache-spark2.4/Portfile

```diff
@@ -0,0 +1,113 @@
+    and general-purpose cluster computing system. It provides high-level APIs in Java, \
+    Scala and Python, and an optimized engine that supports general execution graphs. \
+    It also supports a rich set of higher-level tools including Spark SQL for SQL and \
+    structured data processing, MLlib for machine learning, GraphX for graph processing, and Spark Streaming.
+homepage            https://spark.apache.org/
+platforms           darwin
+supported_archs     noarch
+license             Apache-2
+use_xcode           no
+
+master_sites        http://apache.mirrors.hoobly.com/spark/spark-${version}/ \
+                    http://apache.mirrors.pair.com/spark/spark-${version}/ \
+                    http://apache.spinellicreations.com/spark/spark-${version}/ \
+                    http://mirror.cc.columbia.edu/pub/software/apache/spark/spark-${version}/ \
+                    http://mirror.cogentco.com/pub/apache/spark/spark-${version}/ \
+                    http://mirror.metrocast.net/apache/spark/spark-${version}/ \
+                    http://mirrors.advancedhosters.com/apache/spark/spark-${version}/ \
+                    http://mirrors.ibiblio.org/apache/spark/spark-${version}/ \
+                    http://www.gtlib.gatech.edu/pub/apache/spark/spark-${version}/ \
+                    http://www.trieuvan.com/apache/spark/spark-${version}/ \
+                    https://apache.claz.org/spark/spark-${version}/ \
+                    https://apache.cs.utah.edu/spark/spark-${version}/ \
+                    https://apache.mirrors.lucidnetworks.net/spark/spark-${version}/ \
+                    https://apache.osuosl.org/spark/spark-${version}/ \
+                    https://ftp.wayne.edu/apache/spark/spark-${version}/ \
+                    https://mirror.olnevhost.net/pub/apache/spark/spark-${version}/ \
+                    https://mirrors.gigenet.com/apache/spark/spark-${version}/ \
+                    https://mirrors.koehn.com/apache/spark/spark-${version}/ \
+                    https://mirrors.ocf.berkeley.edu/apache/spark/spark-${version}/ \
+                    https://mirrors.sonic.net/apache/spark/spark-${version}/ \
+                    https://us.mirrors.quenda.co/apache/spark/spark-${version}/ \
+                    https://archive.apache.org/dist/spark/spark-${version}/
+distname            spark-${version}-bin-hadoop2.7
+distfiles           ${distname}.tgz
+extract.suffix      .tgz
+checksums           md5    76afb611aaac5721c9fa91fdc9defa99\
+                    sha1   75b755b8fe55404593dbd1f5069c605ee89b6ab6\
+                    rmd160 a611dafda70c099f8fc57c1df7db73ec9555f419\
+                    sha256 13098490936c9931beda3acc4c30cdc5ca707acd1415eebde1030b11903934fe\
+                    size   233333392
+
+use_configure       no
+
+# Require java version
+java.version        8+
+# JDK port to install if required java not found
+java.fallback       openjdk8
+
+depends_run         port:sbt port:scala2.12 port:python37
```

Comment:
Can it run with python38 instead? That is the current default python version for Macports.

Reply:
Hi Frank. As of right now, from my tests, and from the Spark JIRA issues, PySpark 2.4 does not work with Python 3.8. Please see: https://issues.apache.org/jira/browse/SPARK-29536 where there was an update in October noting that Python 3.8 is not supported yet for Spark 2.4. Here is another issue outlining the problem: https://issues.apache.org/jira/browse/SPARK-31341.
The changes will be in the next/updated PR addressing all your notes. Thank you!